gammapy/irf/psf/psf_map.py

```diff
@@ -80,138 +75,216 @@ def psf_map(self):
     def psf_map(self, value):
         self._irf_map = value
 
-    def get_energy_dependent_table_psf(self, position=None):
-        """Get energy-dependent PSF at a given position.
-
-        By default the PSF at the center of the map is returned.
+    @classmethod
+    def from_geom(cls, geom):
+        """Create psf map from geom.
 
         Parameters
         ----------
-        position : `~astropy.coordinates.SkyCoord`
-            the target position. Should be a single coordinates
+        geom : `Geom`
+            PSF map geometry.
 
         Returns
         -------
-        psf_table : `~gammapy.irf.EnergyDependentTablePSF`
-            the table PSF
+        psf_map : `PSFMap`
+            Point spread function map.
         """
-        if position is None:
-            position = self.psf_map.geom.center_skydir
+        geom_exposure = geom.squash(axis_name="rad")
+        exposure_psf = Map.from_geom(geom_exposure, unit="m2 s")
+        psf_map = Map.from_geom(geom, unit="sr-1")
+        return cls(psf_map, exposure_psf)
 
-        if position.size != 1:
-            raise ValueError(
-                "EnergyDependentTablePSF can be extracted at one single position only."
-            )
+    def to_region_nd_map(self, region):
+        """Convert to region ND PSF map
 
-        energy = self.psf_map.geom.axes["energy_true"].center
-        rad = self.psf_map.geom.axes["rad"].center
+        If a region is given a mean PSF is computed.
 
-        coords = {
-            "skycoord": position,
-            "energy_true": energy.reshape((-1, 1, 1, 1)),
-            "rad": rad.reshape((1, -1, 1, 1)),
-        }
+        Parameters
+        ----------
+        region : `SkyRegion` or `SkyCoord`
+            Region or position where to get the map.
 
-        data = self.psf_map.interp_by_coord(coords)
-        psf_values = u.Quantity(data[:, :, 0, 0], unit=self.psf_map.unit, copy=False)
-
-        if self.exposure_map is not None:
-            coords = {
-                "skycoord": position,
-                "energy_true": energy.reshape((-1, 1, 1)),
-                "rad": 0 * u.deg,
-            }
-            data = self.exposure_map.interp_by_coord(coords).squeeze()
-            exposure = data * self.exposure_map.unit
+        Returns
+        -------
+        psf : `PSFMap`
+            PSF map with region geometry.
+        """
+        if region is None:
```

Comment:
Maybe add a check on the current geometry of the error and raise something if the type is already region?

Reply:
I'd prefer to handle this uniformly with other methods such as `RegionNDMap.get_spectrum()` or similar. I'll add this as a discussion point for the next dev meeting...